lib/mongo/database.rb

```diff
@@ -122,11 +122,13 @@ def collection_names(options = {})
     # @example Get info on each collection.
     #   database.list_collections
     #
-    # @return [ Array<Hash> ] Info for each collection in the database.
+    # @param [ Boolean ] name_only Whether to return only the names of the collections.
+    #
+    # @return [ Array<Hash | String> ] Info for each collection in the database.
     #
     # @since 2.0.5
-    def list_collections
-      View.new(self).list_collections
+    def list_collections(name_only = false)
```

Comment:
I would prefer a `list_collection_names` method which invokes `list_collections` and performs the map to go from hash to string. This will avoid having return value type be parameter dependent.

Reply:
Makes sense; I'll make the change

Reply:
Actually, now that I look at it again, there already is a `collection_names` method; I think we can just close this ticket without needing an changes.

Reply:
Never mind, we do still need to do the change where the `nameOnly` field is sent to the server. I'll revise the change to only do that.

Reply:
Would it be using nameonly though?